katharsis-client/src/main/java/io/katharsis/client/internal/ErrorResponseDeserializer.java

```diff
@@ -0,0 +1,117 @@
+package io.katharsis.client.internal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import io.katharsis.errorhandling.ErrorData;
+import io.katharsis.errorhandling.ErrorResponse;
+
+/**
+ * Serializes top-level Errors object.
+ */
+public class ErrorResponseDeserializer extends JsonDeserializer<ErrorResponse> {
+
+	private static final String LINKS = "links";
+	private static final String ID = "id";
+	private static final String ABOUT_LINK = "about";
+	private static final String STATUS = "status";
+	private static final String CODE = "code";
+	private static final String TITLE = "title";
+	private static final String DETAIL = "detail";
+	private static final String SOURCE = "source";
+	private static final String POINTER = "pointer";
+	private static final String PARAMETER = "parameter";
+	private static final String META = "meta";
+
+	@Override
+	public ErrorResponse deserialize(JsonParser jp, DeserializationContext context)
+			throws IOException, JsonProcessingException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'com.fasterxml.jackson.core.JsonProcessingException' which is a subclass of 'java.io.IOException'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)